File: casadaApi/src/main/java/com/casada/casadaApi/entities/Product.java
package com.casada.casadaApi.entities;

import lombok.Data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serializable;

/**
 * $table.getTableComment()
 */
@Data
@Entity
@Table(name = "product")
public class Product implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "productid", nullable = false)
    private Long productid;

    @Column(name = "productname", nullable = false)
    private String productname;

    @Column(name = "productprice", nullable = false)
    private Double productprice;

    @Column(name = "productactivestatus", nullable = false)
    private Boolean productactivestatus;

    @Column(name = "productcode")
    private String productcode;

}

File: casadaGUI/casada/lib/data/product.dart
class Product {
  int? productId;
  String? productName;
  double? productPrice;
  bool? productActiveStatus;
  String? productCode;

  Product(
      {this.productId,
      this.productName,
      this.productPrice,
      this.productActiveStatus,
      this.productCode});

  Product.fromJson(Map<String, dynamic> json) {
    productId = json['productId'];
    productName = json['productName'];
    productPrice = json['productPrice'];
    productActiveStatus = json['productActiveStatus'];
    productCode = json['productCode'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['productId'] = this.productId;
    data['productName'] = this.productName;
    data['productPrice'] = this.productPrice;
    data['productActiveStatus'] = this.productActiveStatus;
    data['productCode'] = this.productCode;
    return data;
  }
}